testnet-contracts/contracts/Oracle.sol

```diff
@@ -0,0 +1,243 @@
+    */
+    function newOracleClaim(
+        uint256 _bridgeClaimID,
+        bytes32 _message,
+        bytes memory _signature
+    )
+        public
+        onlyValidator
+    {
+        address validatorAddress = msg.sender;
+
+        require(
+            cosmosBridge.isBridgeClaimActive(
+                _bridgeClaimID
+            ) == true,
+            "Can only make oracle claims upon active bridge claims"
+        );
+
+        // Validate the msg.sender's signature
+        require(
+            validatorAddress == valset.recover(
+                _message,
+                _signature
+            ),
+            "Invalid message signature."
+        );
+
+        // Confirm that this address has not already made a claim
+        require(
+            !hasMadeClaim[_bridgeClaimID][validatorAddress],
+            "Cannot make duplicate oracle claims from the same address."
+        );
+
+        hasMadeClaim[_bridgeClaimID][validatorAddress] = true;
+        oracleClaimValidators[_bridgeClaimID].push(validatorAddress);
+
+        emit LogNewOracleClaim(
+            _bridgeClaimID,
+            validatorAddress,
+            _message,
+            _signature
+        );
+    }
+
+    /*
+    * @dev: processProphecyClaim
+    *       Pubically available method which attempts to process a prophecy claim
+    */
+    function processProphecyClaim(
+        uint256 _bridgeClaimID
+    )
+        public
+    {
+        require(
+            cosmosBridge.isBridgeClaimActive(
+                _bridgeClaimID
+            ) == true,
+            "Can only attempt to process active bridge claims"
+        );
+
+        // Process the claim
+        (bool valid,
+            uint256 weightedSignedPower,
+            uint256 weightedTotalPower
+        ) = processClaim(_bridgeClaimID);
+
+        require(
+            valid,
+            "The cumulative power of signatory validators does not meet the threshold"
+        );
+
+        // Update the BridgeClaim's status
+        completeCosmosBridgeClaim(
+            _bridgeClaimID
+        );
+
+        emit LogProphecyProcessed(
+            _bridgeClaimID,
+            weightedSignedPower,
+            weightedTotalPower,
+            msg.sender
+        );
+    }
+
+    /*
+    * @dev: processProphecyClaim
+    *       Operator accessor method which checks if a prophecy claim has passed
+    *       the validity threshold without actually completing the claim
+    */
+    function checkProphecyClaim(
+        uint256 _bridgeClaimID
+    )
+        public
+        view
+        onlyOperator
+        returns(bool, uint256, uint256)
+    {
+        require(
+            cosmosBridge.isBridgeClaimActive(
+                _bridgeClaimID
+            ) == true,
+            "Can only check active bridge claims"
+        );
+        return processClaim(
+            _bridgeClaimID
+        );
+    }
+
+    /*
+    * @dev: processClaim
```

Comment:
hmm the term process to me implies that some completion/mutation may have occured, so could be a little ambiguous. maybe a better name for this, like, checkProphecyThreshold ?

Reply:
https://github.com/cosmos/peggy/pull/82/commits/f8edef9a6846c4f85a67968d0462354719776874